src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -13141,6 +13141,137 @@ BOOL TypeLayoutCheck(MethodTable * pMT, PCCOR_SIGNATURE pBlob)
 
 #endif // FEATURE_READYTORUN
 
+bool IsInstructionSetSupported(CORJIT_FLAGS jitFlags, ReadyToRunInstructionSet r2rInstructionSet)
+{
+    switch (r2rInstructionSet)
+    {
+        case READYTORUN_INSTRUCTION_Aes:
+#if defined(TARGET_X86) || defined(TARGET_AMD64)
+            return jitFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_USE_AES);
```

Comment:
Nit: This whole function might be easier to read if we use two separate `switch`es: one for Intel and one for ARM64.